### PR TITLE
build: Add base IBMz (s390x) support, tested on z15 Debian Buster.

### PR DESCRIPTION
**Pull request by edelsohn**
# Description

This pull request provides basic support to enable oneDNN on IBMz Linux on z (s390x).

Fixes # 817

# Checklist

## Code-change submissions

- [ ] Do all unit and benchdnn tests (`make test` and `make test_benchdnn_*`) pass locally?
- [ ] Have you formatted the code using clang-format?

### New features

- [ ] Have you added relevant tests?
- [ ] Have you provided motivation for adding a new feature?

### Bug fixes

- [ ] Have you added relevant regression tests?
- [ ] Have you included information on how to reproduce the issue (either in a github issue or in this PR)?

## [RFC](https://github.com/oneapi-src/oneDNN/tree/rfcs) PR

- [ ] Does RFC document follow the [template](https://github.com/oneapi-src/oneDNN/blob/rfcs/rfcs/template.md#onednn-design-document-rfc)?
- [ ] Have you added a link to the rendered document?

### CODE REVIEW

**Comment by edelsohn** [comment]
@vpirogov Is there anything that I need to do to make progress with this pull request?

**Comment by emfomenk** [comment]
Hi @edelsohn,

> Is there anything that I need to do to make progress with this pull request?

I guess, just some attention to the PR from the developers :)
Sorry for the delay with review. The patch looks good to me.

File: src/cpu/README.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ enable or disable parts of code. There the following macros defined:
 - `DNNL_X64` is 1 on x64 architecture;
 - `DNNL_AARCH64` is 1 on Arm AArch64 architecture;
 - `DNNL_PPC64` is 1 on OpenPOWER / IBM Power architecture;
+- `DNNL_S390X` is 1 on IBMz / s390x architecture;
```
Comment (vpirogov) [inline src/cpu/README.md:47]: One thing I wanted to confirm is whether `S390X` is the right name for this configuration. Based on my limited knowledge of IBM architecture something like `IBMZ` should be more appropriate.

Comment (edelsohn) [reply src/cpu/README.md:47]: IBMz is the current name of the family of systems that use the Z Architecture, not the name of the architecture itself.  The Architecture now is named z/Architecture. Linux, GNU configure, etc. all use `s390x-ibm-*` as the target triplet.  I will reconfirm the choice, but with the use of s390x throughout the Open Source Software ecosystem, s390x seems more consistent.

Comment (vpirogov) [reply src/cpu/README.md:47]: Thank you for the clarifications!

Comment (edelsohn) [reply src/cpu/README.md:47]: I double-checked and S390X name is fine.

**Comment by emfomenk** [comment]
@edelsohn, thank you for the contribution!

**Pull request merged**